lib/puppet/util/servicenow.rb

```diff
@@ -238,12 +238,14 @@ def selected_facts(settings_hash, facts_query = nil, format = nil)
   end
   module_function :selected_facts
 
-  def event_additional_information
+  def event_additional_information(settings_hash)
     additional_information = {}
     # If we wish to add other top level keys to the additional information field, add them here.
     # Include all facts since this field is not intended for humans.
-    additional_information['facts'] = facts
     additional_information['environment'] = environment
+    selected_facts(settings_hash, nil, :object).map do |key, value|
```

Comment:
btw @gsparks, if you can think of some better merge method than this loop, feel free suggest it because this feels awkward.

Reply:
can we do `selected_facts(settings_hash, nil, :object).merge(additional_information)` ?